### ISSUE TITLE
Exception thrown when textblock text is null and wrap text is selected when creating UI library

### ISSUE THREAD

**Issue by jayrulez** [xenko3d/xenko#82]
System.NullReferenceException: Object reference not set to an instance of an object.
   at Xenko.UI.Controls.TextBlock.UpdateWrappedText(Vector3 availableSpace)
   at Xenko.UI.Controls.TextBlock.MeasureOverride(Vector3 availableSizeWithoutMargins)
   at Xenko.UI.UIElement.Measure(Vector3 availableSizeWithMargins)
   at Xenko.Rendering.UI.UIRenderFeature.DrawInternal(RenderDrawContext context, RenderView renderView, RenderViewStage renderViewStage, Int32 startIndex, Int32 endIndex)
   at Xenko.Rendering.UI.UIRenderFeature.Draw(RenderDrawContext context, RenderView renderView, RenderViewStage renderViewStage, Int32 startIndex, Int32 endIndex)
   at Xenko.Rendering.RenderSystem.Draw(RenderDrawContext renderDrawContext, RenderView renderView, RenderStage renderStage)
   at Xenko.Rendering.Compositing.ForwardRenderer.DrawView(RenderContext context, RenderDrawContext drawContext, Int32 eyeIndex, Int32 eyeCount)
   at Xenko.Rendering.Compositing.ForwardRenderer.DrawCore(RenderContext context, RenderDrawContext drawContext)
   at Xenko.Rendering.Compositing.SceneRendererBase.Draw(RenderDrawContext context)
   at Xenko.Rendering.Compositing.EditorTopLevelCompositor.DrawInner(RenderDrawContext context)
   at Xenko.Rendering.Compositing.SceneCameraRenderer.DrawCore(RenderContext context, RenderDrawContext drawContext)
   at Xenko.Rendering.Compositing.SceneRendererBase.Draw(RenderDrawContext context)
   at Xenko.Rendering.Compositing.GraphicsCompositor.DrawCore(RenderDrawContext context)
   at Xenko.Rendering.RendererBase.Draw(RenderDrawContext context)
   at Xenko.Engine.SceneSystem.Draw(GameTime gameTime)
   at Xenko.Games.GameSystemCollection.Draw(GameTime gameTime)
   at Xenko.Games.GameBase.Draw(GameTime gameTime)
   at Xenko.Editor.EditorGame.Game.EditorServiceGame.Draw(GameTime gameTime)

**Comment by jayrulez** [comment]
This will cause an exception to be thrown when text is null.
https://github.com/xenko3d/xenko/blob/master/sources/engine/Xenko.UI/Controls/TextBlock.cs#L264

Is it okay to make the default value an empty string instead of null?

**Comment by Kryptos-FR** [comment]
I might be wrong but I think there is a limitation in the serialization that makes null and empty string behave the same. And changing default value can have side effects.

The safest way is to check for null at the beginning of the `UpdateWrappedText` method and if it is set `wrappedText` to `null` and return immediately.

**Comment by Kryptos-FR** [comment]
As a side-note on default values. There should be considered as both part of the documentation and part of the API. Hence changing it's value is a breaking change.

Even when the default value was poorly chosen, changing it should be for a good reason. Fixing a mere `NullReferenceException` is not enough in my opinion to justify a breaking change.

I believe we need more test coverage before we can make those kind of changes safely.

**Comment by jayrulez** [comment]
Ok I will go with the first option of returning if the text is null

```C#
            if(text == null)
            {
                if(WrapText)
                {
                    wrappedText = string.Empty;
                }

                return;
            }
```

The condition is there because if WrapText is checked then null will be returned which will cause UIBatch drawstring to throw also.

**Comment by dfkeenan** [comment]
The method `UpdateWrappedText` is only called if `WrapText` is true.  If you are going to add that to the top of that method you don't really need the `if(WrapText)`.

**Comment by Kryptos-FR** [comment]
@dfkeenan Good point. I already mentioned it in the pull-request #84 :smile:

**Comment by dfkeenan** [comment]
I didn't see the pull request while I was in here. I saw you had similar comments. 😄 

Things are really active around here now. You have to be quick.

**Comment by jayrulez** [comment]
Okay. I updated the pull request. Thanks for the feedback.

**Comment by xen2** [comment]
Fixed with #84
